### PR TITLE
add client streaming response trace and prefix all traces with "sent:" or "recv:"

### PR DESCRIPTION
**Pull request by yangzhouhan**

### CODE REVIEW

**Comment by dsymonds** [comment]
The commit description should also mention that it is prefixing all the traces with "sent: " or "received: ".

File: call.go [hunk 1]
```diff
@@ -184,7 +183,7 @@ func Invoke(ctx context.Context, method string, args, reply interface{}, cc *Cli
 			continue
 		}
 		if EnableTracing {
-			c.traceInfo.tr.LazyLog(payload{reply}, true)
+			c.traceInfo.tr.LazyLog(&fmtStringer{"received: %v", []interface{}{payload{reply}}}, true)
```
Comment (dsymonds) [inline call.go]: please use "recv: %v" instead. it'll line up better with "sent: %v".

Comment (dsymonds) [inline call.go]: better yet, put a little bool in the payload struct and render the prefix lazily too.

**Comment by yangzhouhan** [comment]
done. PTAL

File: call.go [hunk 2]
```diff
@@ -165,7 +164,11 @@ func Invoke(ctx context.Context, method string, args, reply interface{}, cc *Cli
 			return toRPCErr(err)
 		}
 		if EnableTracing {
-			c.traceInfo.tr.LazyLog(payload{args}, true)
+			p := &payload{
```
Comment (dsymonds) [inline call.go]: there's no need for these to be separate vars. they are tiny. inline it in the LazyLog calls.

File: stream.go [hunk 3]
```diff
@@ -195,6 +199,19 @@ func (cs *clientStream) RecvMsg(m interface{}) (err error) {
 		}
 	}()
 	if err == nil {
+		if cs.tracing {
+			cs.mu.Lock()
+			if cs.traceInfo.tr != nil {
+				if cs.traceInfo.tr != nil {
```
Comment (dsymonds) [inline stream.go]: duplicate nil check

File: trace.go [hunk 4]
```diff
@@ -93,12 +93,17 @@ func (f *firstLine) String() string {
 
 // payload represents an RPC request or response payload.
 type payload struct {
-	m interface{} // e.g. a proto.Message
+	sent bool        // whether this is a request or response
```
Comment (dsymonds) [inline trace.go]: // whether this is an outgoing payload

**Comment by yangzhouhan** [comment]
fixed

**Comment by dsymonds** [comment]
LGTM

**Pull request merged**